packages/backend/src/services/RolesService/RolesService.ts

```diff
@@ -859,4 +861,80 @@ export class RolesService extends BaseService {
             },
         });
     }
+
+    async duplicateRole(
+        account: Account,
+        organizationUuid: string,
+        roleUuid: string,
+        duplicateRoleData: CreateRole,
+    ): Promise<RoleWithScopes> {
+        RolesService.validateOrganizationAccess(account, organizationUuid);
+
+        const { name, description } = duplicateRoleData;
+        if (name !== undefined) {
```

Comment:
should we throw error if name === undefined ? or `RolesService.validateRoleName(name || '');`

Reply:
Done